lib/ngx/re.lua

```diff
@@ -84,6 +84,9 @@ end
 
 
 function _M.split(subj, regex, opts, ctx, max, res)
+    if subj == nil then
+        return nil, nil
```

Comment:
> you should really throw an error here instead instead of passing along nil.

I feel like this would be a good behavior in such a case. The same way the Lua `string` API and the `ngx.re` API error out when a given subject is `nil`, we should probably also behave the same way. Shouldn't we?